includes/libraries/class-wc-eval-math.php

```diff
@@ -282,24 +290,57 @@ private static function pfx( $tokens, $vars = array() ) {
 							}
 							$stack->push( $op1 / $op2 );
 							break;
+						case '%':
+							if ( 0 == $op2 ) {
+								return self::trigger( 'division by zero' );
+							}
+							$stack->push( $op1 % $op2 );
+							break;
+						case '\\':
+							if ( 0 == $op2 ) {
+								return self::trigger( 'division by zero' );
+							}
+							$stack->push( intdiv( $op1, $op2 ) );
+							break;
 						case '^':
 							$stack->push( pow( $op1, $op2 ) );
 							break;
 					}
 					// if the token is a unary operator, pop one value off the stack, do the operation, and push it back on
 				} elseif ( '_' === $token ) {
 					$stack->push( -1 * $stack->pop() );
-					// if the token is a function, pop arguments off the stack, hand them to the function, and push the result back on
+					// if the token is a number, convert it and push
+					// if the token is a variable, push it's value
 				} elseif ( ! preg_match( "/^([a-z]\w*)\($/", $token, $matches ) ) {
 					if ( is_numeric( $token ) ) {
-						$stack->push( $token );
+						$stack->push( $token + 0 );
```

Comment:
I added this to make sure that only numbers are on the stack. While this is normally not needed (operations convert automatically), e.g. `evaluate("2")` unexpectedly returned a string.

Reply:
`$token + 0` will probably give a warning in PHP 7.4 when `$token` is a string. It would be better to cast `$token` to an `int` or `float`.

Reply:
`$token` was always a string, the only thing that changed is that I explicitly do it here instead of above where the operators are evaluated, so the warning was there before if there was one.
So unconditionally convert to float? Or should I try to convert to int if possible else float?